Library/Homebrew/formula.rb

```diff
@@ -1582,6 +1582,21 @@ def time
     end
   end
 
+  sig { params(targets: T.nilable(T.any(Pathname, String))).void }
+  def deuniversalize_machos(*targets)
+    if targets.blank?
+      targets = any_installed_keg.mach_o_files.select do |file|
+        file.arch == :universal && file.archs.include?(Hardware::CPU.arch)
+      end
+    end
+
+    targets.each do |t|
+      macho = MachO::FatFile.new(t)
+      native_slice = macho.extract(Hardware::CPU.arch)
+      native_slice.write t
```

Comment:
I'll probably need to make sure that `t` is writeable here, and handle load failures (what if `t` is not a `FatFile`?) or extraction failures (what if `t` does not have a `Hardware::CPU.arch` slice?).